pkgs/applications/editors/neovim/tests/default.nix

```
@@ -260,6 +260,21 @@ pkgs.recurseIntoAttrs (rec {
     [ "$result" = 0 ]
   '';
 
+  # Generate a neovim wrapper with only a init.lua and no init.vim file
+  nvim_with_only_init_lua = wrapNeovim2 "-only-lua-init-file" {
+    luaRcContent = "-- some text";
+  };
+
+  # check that we do not generate an init.vim file if it is not needed
+  no_init_vim_file = runTest nvim_with_only_init_lua ''
+    ${nvim_with_only_init_lua}/bin/nvim -i NONE -e --headless -c 'if len(getscriptinfo({"name":"init.vim"})) == 0 | quit | else | cquit | fi'
+    # This does now work because the lua file is sourced via loadfile() which
+    # does not add the file name to :scriptnames and getscriptinfo().
+    #${nvim_with_only_init_lua}/bin/nvim -i NONE -e --headless -c 'if len(getscriptinfo({"name":"init.lua"})) == 1 | quit | else | cquit | fi'
+
+    grep 'VIMINIT=.*init.lua' ${nvim_with_only_init_lua}/bin/nvim
```

```suggestion
    assertFileRegex ${nvim_with_only_init_lua}/bin/nvim 'VIMINIT=.*init.lua'
```

Use our testing utilities at our disposal for more helpful logging messages on failure. 

>test-neovim> Expected /nix/store/1pnhd04xd5fraxxh3khxl3x5jvdj38kn-neovim-0.11.2-only-lua-init-file/bin/nvim to match VIlMINIT=.*init.lua but it did not.
error: builder for '/nix/store/8f2k4y34d31pfjg0b79ckwb6263gqcvn-test-neovim-0.11.2-only-lua-init-file.drv' failed with exit code 1;
       last 1 log lines:
       > Expected /nix/store/1pnhd04xd5fraxxh3khxl3x5jvdj38kn-neovim-0.11.2-only-lua-init-file/bin/nvim to match VIlMINIT=.*init.lua but it did not.
       For full logs, run:
         nix log /nix/store/8f2k4y34d31pfjg0b79ckwb6263gqcvn-test-neovim-0.11.2-only-lua-init-file.drv